pipelines/tekton/gitops-update-pipeline/tasks/yq-update.yaml

```diff
@@ -0,0 +1,51 @@
+# Adapted from:
+# https://github.com/tektoncd/catalog/blob/main/task/yq/0.4/yq.yaml
+#
+# If we decide to pass image details to pipeline rather than determining them
+# from existing PipelineRuns, then we could use that task directly.
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: yq-update
+spec:
+  description: >-
+    This task is used by the GitOps update Pipeline, to allow modifying files with yq
+  workspaces:
+    - name: source
+      description: A workspace that contains the files which needs to be altered.
+  params:
+    - name: SCRIPT
+      type: string
+      description: The yq script to execute. Can be multiple lines for complex tasks.
+      default: ""
+    - name: image
+      type: string
+      description: The yq image to use.
+      default: docker.io/mikefarah/yq:4.27.5@sha256:2be3626ed633fbe1fc33ee9343a1256a6be53334412b2251b9a859f8c145bb53
```

Comment:
Once this goes to the product, we will need to productize it, won't we? Right now there doesn't seem to be yq image at [RH catalog](https://catalog.redhat.com/software/containers/search).

Reply:
Yes, I guess so. I haven't thought much about that. If there end up being other tools that we need in tekton tasks, then maybe we could have some sort of `odh-tekton-tools` image containing them all, rather than needing a different image for each